c++/src/Statistics.hh

```diff
@@ -41,49 +41,181 @@ namespace orc {
   };
 
 /**
+ * Internal Statistics Implementation
+ */
+
+  template <typename T>
+  class InternalStatisticsImpl {
+  private:
+    bool hasNull_;
```

Comment:
Personally I don't have any problem with the current style. I don't even notice the inconsistency (should have pay more attentions to style, :)). But to me, missing writer is a big functionality gap that needs to be filled in quickly.

Having said that I think this is eventually a call from @omalley and other major committers. If we really want to enforce a new style, we should always update the existing code first before adding new codes with new style. Otherwise the code base is going to look even worse until we fix everything.